reframe/frontend/executors/policies.py

```diff
@@ -405,25 +417,14 @@ def advance_wait(self, task):
             # Not all dependencies have finished yet
             return 0
 
-    def advance_ready_to_compile(self, task):
-        if task.check.local or task.check.build_locally:
-            if len(self._local_scheduler_tasks) < self._rfm_max_jobs:
-                try:
-                    task.compile()
-                    task.policy_stage = 'compiling'
-                    self._local_scheduler_tasks.add(task)
-                except TaskExit:
-                    self._current_tasks.remove(task)
-
-                return 1
-            else:
-                return 0
-
-        partname = task.check.current_partition.fullname
+    def advance_setup(self, task):
+        partname = (
+            '_rfm_local' if task.check.local or task.check.build_locally
+            else task.check.current_partition.fullname
+        )
```

Comment:
Make that a utility function at the top of the file:

```suggestion
        partname = _get_partition_name(task.check)
```